### PR TITLE
Update haskell.nix to get latest ghcjs emcc fixes

### PR DESCRIPTION
**Pull request by hamishmack**
### Jira ticket

ADP-787

### Overview

This uses the PR input-output-hk/haskell.nix#1064 and makes a working ghcjs build.

emcc is the emscriten C to JS compiler used to by ghcjs when running hsc2hs because so that constants in C code compiled with emcc will match constants in the code generated by ghcjs.

### Comments

```screen
$ nix-build release.nix -A ghcjs.cardano-address.x86_64-linux
$ node result/bin/cardano-address.jsexe/all.js
Usage: all.js ((-v|--version) | COMMAND)
  Command-line for address and key manipulation in Cardano.
...

$ nix-build release.nix -A ghcjs.haskellPackages.cardano-addresses.checks.unit.x86_64-linux

```

[Hydra jobset](https://hydra.iohk.io/jobset/Cardano/cardano-addresses-pr-108#tabs-jobs)

### CODE REVIEW

**Review by rvl — COMMENTED** [review]
Should we hold this open until the Haskell.nix PR is merged?

What do we need emcc for? Is it feasible to javascript-compile all the crypto C code through nix without using pre-compiled javascript blobs?

Would you mind updating the PR description and fill in the `<blanks>`?

**Comment by hamishmack** [comment]
> Should we hold this open until the Haskell.nix PR is merged?

There were some GHCJS on darwin test failures on hydra related eval running on linux.  Those should be resolved now and once the tests pass we can merge the haskell.nix PR.

> What do we need emcc for? Is it feasible to javascript-compile all the crypto C code through nix without using pre-compiled javascript blobs?

We need it to make sure constants and structure layout is correct between C code compiled with emcc and haskell code that uses hsc2hs.

> Would you mind updating the PR description and fill in the <blanks>?

Done

**Comment by rvl** [comment]
Thanks, awesome! :tada:

**Comment by rvl** [comment]
Nice. I looked at the Enum fix and it seems generally good. So if we update and review #109 first, then we can merge this.

**Review by rvl — APPROVED** [review]
This build works great!
I tested on top of #112.

**Pull request merged**
